receiver/k8sclusterreceiver/receiver.go

```diff
@@ -85,11 +84,12 @@ func (kr *kubernetesReceiver) Shutdown(context.Context) error {
 
 func (kr *kubernetesReceiver) dispatchMetricData(ctx context.Context) (numTimeseries int, numPoints int, errs []error) {
 	for _, m := range kr.resourceWatcher.dataCollector.CollectMetricData() {
+		numTs, numPts := obsreport.CountMetricPoints(m)
+		numTimeseries += numTs
+		numPoints += numPts
+
```

Comment:
Why moving this out? Looks like we are including metric that were failed to be sent as well

Reply:
Looks like we need to move obsreport.StartMetricsReceiveOp/obsreport.EndMetricsReceiveOp inside this function to correctly separate metrics that were successfully sent or failed. Otherwise, if we have part of the metrics failed to be sent and another succeeded, we cannot mark them correctly.

Reply:
Apparently, that is handled here: https://github.com/open-telemetry/opentelemetry-collector/blob/master/obsreport/obsreport_receiver.go#L299. 

Right now, MTSes are only counted if `ConsumeMetricData` fails. However, we want to always count it. And the the logic in `obsreport` (in the link above) is expected to do the right thing, i.e. count MTSes towards `otelcol_receiver_accepted_metric_points` if there's no error and if there's an error the count will be accounted for in `otelcol_receiver_refused_metric_points`.

Reply:
Missed your second comment earlier. That's a good point. Updated to capture the metric counts and errors per invocation of `ConsumeMetricsData`.